src/machines/java/com/enderio/machines/common/blockentity/PoweredSpawnerBlockEntity.java

```diff
@@ -0,0 +1,126 @@
+package com.enderio.machines.common.blockentity;
+
+import com.enderio.api.capability.StoredEntityData;
+import com.enderio.api.capacitor.CapacitorModifier;
+import com.enderio.api.capacitor.QuadraticScalable;
+import com.enderio.base.common.particle.RangeParticleData;
+import com.enderio.core.common.sync.BooleanDataSlot;
+import com.enderio.core.common.sync.SyncMode;
+import com.enderio.machines.common.blockentity.base.PoweredTaskMachineEntity;
+import com.enderio.machines.common.blockentity.task.SpawnTask;
+import com.enderio.machines.common.io.item.MachineInventoryLayout;
+import com.enderio.machines.common.menu.PoweredSpawnerMenu;
+import net.minecraft.core.BlockPos;
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.server.level.ServerLevel;
+import net.minecraft.server.level.ServerPlayer;
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.EntityType;
+import net.minecraft.world.entity.player.Inventory;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.inventory.AbstractContainerMenu;
+import net.minecraft.world.level.block.entity.BlockEntityType;
+import net.minecraft.world.level.block.state.BlockState;
+import net.minecraft.world.phys.Vec3;
+import net.minecraftforge.registries.ForgeRegistries;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Objects;
+
+public class PoweredSpawnerBlockEntity extends PoweredTaskMachineEntity<SpawnTask> {
+
+    public static final QuadraticScalable CAPACITY = new QuadraticScalable(CapacitorModifier.ENERGY_CAPACITY, () -> 100000f);
+    public static final QuadraticScalable TRANSFER = new QuadraticScalable(CapacitorModifier.ENERGY_TRANSFER, () -> 200f);
+    public static final QuadraticScalable USAGE = new QuadraticScalable(CapacitorModifier.ENERGY_USE, () -> 160f);
+    private StoredEntityData entityData = StoredEntityData.empty();
+    private int range = 3;
+    private boolean rangeVisible;
+    protected float rCol = 1;
+    protected float gCol = 0;
+    protected float bCol = 0;
```

Comment:
I'm planning on bringing all the range methods to the base machine class in a next PR. Should these colours be in a config, they might have been in the past, but does it have an advantage? Does it even help colour blind people?

Reply:
The config right now resides on the server side and thus not very useful as the colors will be synced to all players. It can be helpful but we need to move it client side for any use.  If we do decide to not make it configurable, we will need to find a good default color that works for all